rosetta/retriever/retriever_test.go

```diff
@@ -551,10 +602,14 @@ func TestRetriever_Block(t *testing.T) {
 			return []flow.Event{}, nil
 		}
 
-		ret, err := baselineRetriever(t)
-		require.NoError(t, err)
-
-		ret.index = index
+		ret := retriever.New(
+			mocks.GenericParams,
+			index,
+			mocks.BaselineValidator(t),
+			mocks.BaselineGenerator(t),
+			mocks.BaselineInvoker(t),
+			mocks.BaselineConverter(t),
+		)
```

Comment:
You do this approximately _a million_ times in these tests. Why not put it into a helper function, something like this:

```go
func baselineRetriever(t, options ...func(*Dependencies) {
  deps := Dependencies {
    index:        mocks.BaselineIndex(t),
    validator: mocks.BaselineValidator(t),
    ..
  }
  for _, option := range options {
    option(deps)
  }
  ret := retriever.New(dps.params, deps.index, deps.validator, ...)
}

func WithGenerator(gen Generator) func(*Retriever) {
  return func(deps *Dependencies) {
    deps.gen = gen
  }
}
```

Reply:
Then, all of the initialialization turns into one-liners like this:

```go
ret := baselineRetriever(t, WithGenerator(gen))
```

Reply:
Yep I thought about this, but thought it's maybe not done often enough to warrant that kind of func. If you also think it is worth it though, sure, can do! It will make stuff more readable for sure 👍

Reply:
If you haven't done it already, I'm OK not doing it now, but definitely seems like it's warranted with how often it is repeated. Copy-pasting such stuff is always fine once everything passes, but then it leaves the door open for errors due to lack of attention when expanding tests later.